### PR TITLE
[REF] point_of_sale*: remove dead code

### PR DESCRIPTION
**Pull request by vlst-odoo**
It turns out that there are many functions in the pos codebase that are never used.

In this commit we remove them.

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/point_of_sale/static/src/app/store/models.js [hunk 1]
```diff
@@ -2463,9 +2360,6 @@ export class Order extends PosModel {
     set_screen_data(value) {
         this.screen_data["value"] = value;
     }
-    get_current_screen_data() {
-        return this.screen_data["value"] ?? { name: "ProductScreen" };
-    }
```
Comment (adgu-odoo) [inline addons/point_of_sale/static/src/app/store/models.js:2468]: I think this is used in the customer display template

File: addons/point_of_sale/static/src/app/store/models.js [hunk 2]
```diff
@@ -1792,10 +1742,6 @@ export class Order extends PosModel {
         return true;
     }
 
-    isFirstDraft() {
```
Comment (adgu-odoo) [inline addons/point_of_sale/static/src/app/store/models.js:1795]: I think you can also remove all occurence of firstDraft in the code

**Review by adgu-odoo — CHANGES_REQUESTED** [review]

File: addons/point_of_sale/static/src/app/store/models.js (hunk 1 above)
Comment (vlst-odoo) [reply addons/point_of_sale/static/src/app/store/models.js:2468]: good catch! I added it back

File: addons/point_of_sale/static/src/app/store/models.js (hunk 2 above)
Comment (vlst-odoo) [reply addons/point_of_sale/static/src/app/store/models.js:1795]: :+1:

**Review by adgu-odoo — APPROVED** [review]

**Comment by adgu-odoo** [comment]
@robodoo r+

**Pull request closed without merging**